data-collection/deploy/module-trusted-advisor.yaml

```
@@ -224,6 +232,136 @@ Resources:
+                  return response.get('recommendation', {})
+              except trustedadvisor.exceptions.ClientError as e:
+                  print(f"Error getting recommendation details for {recommendation_arn}: {str(e)}")
+                  raise
+      
+          def get_resource_summaries(trustedadvisor,recommendation_id: str) -> List[Dict]:
+              """
+              Fetch resource summaries for a given recommendation ID using pagination
+              """
+              resources = []
+              next_token = None
+              try:
+                  while True:
+                      params = {'recommendationIdentifier': recommendation_id}
+                      if next_token:
+                          params['nextToken'] = next_token
+                      
+                      response = trustedadvisor.list_recommendation_resources(**params)
+                      
+                      if 'recommendationResourceSummaries' in response:
+                          resources.extend(response['recommendationResourceSummaries'])
+                      
+                      if 'nextToken' not in response:
+                          break
+                          
+                      next_token = response['nextToken']
+                      
+                  return resources
+              
+              except trustedadvisor.exceptions.ClientError as e:
+                  print(f"Error fetching resource summaries for recommendation {recommendation_id}: {str(e)}")
+                  raise
+
+          def write_to_file(account_id, account_name, payer_id, recommendations: List[Dict]):
+              """
+              Write recommendations and their resources to a file
+              """
+              trustedadvisor = assume_role(account_id, "trustedadvisor", REGIONS[0], ROLE_NAME)
+              try:
+                  with open(TMP_FILE_Priority, 'w', encoding='utf-8') as jsonfile:
+                      for recommendation in recommendations:
+                          rec_id = recommendation['id']
+                          rec_creation_date = recommendation['createdAt'].isoformat() if isinstance(recommendation['createdAt'], datetime) else 'N/A'
+                          rec_lastupdated_date = recommendation['lastUpdatedAt'].isoformat() if isinstance(recommendation['lastUpdatedAt'], datetime) else 'N/A'
+                          
+                          # Get recommendation details including resolved date
+                          recommendation_details = get_recommendation_details(trustedadvisor, recommendation['arn'])
+                          rec_resolved_date = recommendation_details['resolvedAt'].isoformat() if isinstance(recommendation_details['resolvedAt'], datetime) else 'N/A'
+
+                          # Get resources for this recommendation
+                          try:
+                              resources = get_resource_summaries(trustedadvisor, recommendation['arn'])
```

you can probably just use paginator instead of get_resource_summaries function
```suggestion
                              resources = (trustedadvisor
                                     .get_paginator('list_recommendation_resources')
                                     .paginate(recommendationIdentifier=recommendation['arn'])
                                     .search('recommendationResourceSummaries')
                              )
```